### ISSUE TITLE
nfts().findAllByCreator() - Nondeterministic responses 

### ISSUE THREAD

**Issue by yintothayang** [metaplex-foundation/js#196]
The [findAllByCreator()](https://github.com/metaplex-foundation/js#findAllByCreator) function seemingly has nondeterministic responses. 

On multiple executions of 
```
  const owner_nfts = await metaplex.nfts().findAllByOwner(new PublicKey('ACt2d7CMq2peS35GJuzWqn8ACbbVJqz326iiPn1mPaXD')).run();
  const creator_address = owner_nfts[0].creators[0].address
  const creator_nfts = await metaplex.nfts().findAllByCreator(creator_address).run()
```
I will either get the correct response (an array of 2,500 NFT objects), or an empty array. 

It seems like there might be a timeout getting hit, because of the large number of NFTs. Unfortunately this doesn't throw, it just returns an empty array. 

owner:
https://solana.fm/address/ACt2d7CMq2peS35GJuzWqn8ACbbVJqz326iiPn1mPaXD

creator:
https://solana.fm/address/9yz273zB6rQHyptbSpVvC75o4G17NwJrTk4u2ZiNV3tZ

**Comment by yintothayang** [comment]
Is there anything else I could provide, so this issue gets addressed?

**Comment by yintothayang** [comment]
But this function isn't throwing a timeout error, it's just returning an incorrect response? 
I don't think it's a matter of "inefficiency", if the RPC call times out that's fine, but it shouldn't just return an empty array.

**Comment by lorisleiva** [comment]
If you check the HTTP request/response, is the RPC returning the empty array?

**Comment by yintothayang** [comment]
Hi @lorisleiva , seems like this could be the source of my troubles -> https://github.com/metaplex-foundation/js/blob/31a76e2bec0b000fdca6616a7bbe65ba806864c3/packages/js/src/utils/GmaBuilder.ts#L118
Instead of throwing, say if the RPC call timedout, you assume the addresses don't exist, and return an array where `exists == false`.

Further up the chain this becomes a problem -> https://github.com/metaplex-foundation/js/blob/31a76e2bec0b000fdca6616a7bbe65ba806864c3/packages/js/src/plugins/nftModule/findNftsByMintList.ts#L66
Now that the accounts have been maked as `exists == false`, will produce an empty array.


I have written a small test for you (trying to keep the same function names)
```
async function getChunks(){
  const chunks = [['1', '2'], ['3', '4'], ['5']]

  const chunkPromises = chunks.map((chunk)=>getChunk(chunk))
  const resolvedChunks = await Promise.allSettled(chunkPromises)

  console.log("resolvedChunks", resolvedChunks)
  return resolvedChunks.flatMap((result)=>result.status === 'fulfilled' ? result.value : [])
}

async function getChunk(publicKeys){
  try {
    // TODO(loris): Use lower level RPC call to add dataSlice support.
    return await RPCTimeout()
  } catch (error) {
    // TODO(loris): Throw error instead?
    return publicKeys.map((publicKey) => ({
      publicKey: publicKey,
      exists: false,
    }));
  }
}

async function RPCTimeout(){
  await new Promise(resolve => setTimeout(resolve, 100))
  throw "Timeout"
}

async function main(){
  const res = await getChunks()
  console.log(res)
}

main()
```

**Comment by yintothayang** [comment]
Could you reopen this issue?

**Comment by lorisleiva** [comment]
Thanks for detailing your findings, that really helped! I know how to fix this, I'll reopen and tackle this as soon as I can.

**Comment by yintothayang** [comment]
awesome, thank you!

**Comment by lorisleiva** [comment]
This should now be fixed in version [`0.15`](https://github.com/metaplex-foundation/js/releases/tag/js%2Fv0.15.0). 🚀
